assets/blocks/quiz/answer-blocks/index.js

```diff
@@ -84,15 +100,28 @@ const questionTypes = {
 		settings: [],
 		validate: ( { before, after, gap } = {} ) => {
 			return {
-				noGap: ! gap?.length,
-				noBefore: ! before,
-				noAfter: ! after,
+				noGap: ! gap?.filter( ( val ) => val !== '' ).length,
+				noGapWhitespace: ! gap?.filter( ( val ) => val.trim() !== '' )
+					.length,
+				noBeforeAndNoAfter: ! before && ! after,
+				noBeforeAndNoAfterWhitespace:
+					! before?.trim() && ! after?.trim(),
 			};
 		},
 		messages: {
 			noGap: __( 'Add a right answer to this question.', 'sensei-lms' ),
-			noBefore: __( 'Add text before and after the gap.', 'sensei-lms' ),
-			noAfter: __( 'Add text before and after the gap.', 'sensei-lms' ),
+			noGapWhitespace: __(
```

Comment:
I'm partial to simplifying gap fill too and not having a separate message for whitespace characters.